### PR TITLE
feat: remove `SilentTQDM`.

### PR DESCRIPTION
**Pull request by tm91236**
### PR Type
- Feature
- Refactoring (no functional changes)
- Documentation content changes
- Tests

### Description
To reduce the number of lines of code, and subsequent complexity throughout the codebase, the `SilentTQDM` class has been removed without loss of functionality.

The `SilentTQDM` functionality is now provided through a boolean `progress_bar` variable and silencing performed via the `disable` keyword argument to `tqdm.tqdm`. The `write` method is silenced by making all calls to the method conditional on the value of the `progress_bar` boolean.

### How Has This Been Tested?
Existing tests pass as expected.

### Does this PR introduce a breaking change?
`SilentTQDM` will be removed. No deprecation is required as this was an internal part of the API.
It may be good to be more explicit about what parts of the API are internal/external to enable more flexibility and allow for faster deprecation where appropriate.

### Checklist before requesting a review
- [ ] I have made sure that my PR is not a duplicate.
- [ ] My code follows the style guidelines of this project.
- [ ] I have ensured my code is easy to understand, including docstrings and comments where necessary.
- [ ] I have performed a self-review of my code.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have updated CHANGELOG.md, if appropriate.

### CODE REVIEW

**Review by gchqdev227 — CHANGES_REQUESTED** [review]
Some things to clarify, but no overall concerns with this being removed.  Benefit to using existing `tdqm` functionality.

In theory no performance hit – I'd be slightly concerned about the additional check for progress bar in the `if` statement below, but it doesn't appear to have degraded the performance. Having said that, it might be worth considering a refactor in future, as in theory we can remove that `if` statement altogether in cases of `progress=False`? Not sure how important that would be though.

Looks like code coverage has slipped. Perhaps add some `pass` statements to bring the number back up? (joke)

File: coreax/score_matching.py [hunk 1]
```diff
@@ -504,8 +502,8 @@ def match(self, x):  # noqa: C901, PLR0912
             state, val = train_step(state, x[idx, :], random_vectors[idx, :])
 
             # Print progress (limited to avoid excessive output)
-            if i % 10 == 0:
-                self.progress_bar.write(f"{i:>6}/{self.num_epochs}: loss {val:<.5f}")
+            if i % 10 == 0 and self.progress_bar:
```
[the remark this answers is not in the document]
Comment (gchqdev227) [reply coreax/score_matching.py:505]: > I guess it would be better to rewrite this as if self.progress_bar and i%10 == 0 so we only compute i%10 if the progress bar is enabled.
Good catch!
> Without this check, the disabled progress bar would still write this output while the prior SilentTQDM did not.
So even if the progress bar is disabled, calling `write` still prints to stdout/stderr?

[the remark this answers is not in the document]
Comment (tm91236) [reply coreax/score_matching.py:505]: Yes, in the following MWE the progress bar is disabled but will still write to stdout on every 10th step.
```python
import time

from tqdm import tqdm

N = 50
progress_bar = False
tqdm_progress_bar = tqdm(range(N), disable=not progress_bar)

for i in tqdm_progress_bar:
    time.sleep(0.1)
    if i % 10 == 0:
        tqdm_progress_bar.write(f"Intermediate Result {i}/{N}")

# STDOUT (progress_bar=False):
# Intermediate Result 0/50
# Intermediate Result 10/50
# Intermediate Result 20/50
# Intermediate Result 30/50
# Intermediate Result 40/50
```

[the remark this answers is not in the document]
Comment (gchqdev227) [reply coreax/score_matching.py:505]: Interesting!

File: tests/unit/test_util.py
[the remark this answers is not in the document]
Comment (gchqdev227) [reply tests/unit/test_util.py:1]: I think we're using a sufficiently obvious part of the `tqdm ` API that we can afford not to have explicit tests. If we end up refactoring to avoid invoking the progress bar or loop completely, then I imagine a test would be in order.

**Comment by gchqdev227** [comment]
I'm approving despite coverage decrease, as it's a tiny amount only due to the removal of lines overall.
